src/option.cc

```diff
@@ -132,7 +132,8 @@ void process_environment(const char ** envp, const string& tag,
         if (*q == '_')
           *r++ = '-';
         else
-          *r++ = static_cast<char>(std::tolower(*q));
+          *r++ = static_cast<char>(std::tolower(
+            static_cast<unsigned char>(*q)));
```

Comment:
Is first casting to `unsigned char` and then casting to `char` intentional here? If yes, what are the implications?

Reply:
Thanks for taking the time to reply in detail, @riastradh. I agree with your reasoning.